src/test/java/com/jcabi/log/SupplierLoggerTest.java

```diff
@@ -0,0 +1,278 @@
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.log;
+
+import org.apache.log4j.Level;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link SupplierLogger}.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.18
+ */
+@SuppressWarnings("PMD.MoreThanOneLogger")
+public final class SupplierLoggerTest {
+
+    /**
+     * SupplierLogger can tell if debug is disabled and the message is not
+     * logged enabled.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void debugIsDisabled() throws Exception {
+        final String loggerName = "nodebug";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(loggerName);
+        final UnitTestAppender app = new UnitTestAppender();
+        app.setName("nodebugapp");
+        app.activateOptions();
+        logger.addAppender(app);
+        logger.setLevel(Level.ERROR);
+        Logger.withSupplier().debug(
+            loggerName, "Debug disabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return "test1";
+                }
+            }
+        );
+        MatcherAssert.assertThat(
+            new String(app.output().toByteArray()),
+            Matchers.isEmptyString()
+        );
+    }
+
+    /**
+     * SupplierLogger can log a message with debug level.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void debugIsEnabled() throws Exception {
+        final String loggerName = "debugen";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(loggerName);
+        final UnitTestAppender app = new UnitTestAppender();
+        app.setName("debugapp");
+        app.activateOptions();
+        logger.addAppender(app);
```

Comment:
@amihaiemil I think that logger creation and adding appender could also be extracted to separate method